src/Annotation/ApiResource.php

```diff
@@ -13,13 +13,44 @@
 
 namespace ApiPlatform\Core\Annotation;
 
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\Common\Util\Inflector;
+
 /**
  * ApiResource annotation.
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
  *
  * @Annotation
  * @Target({"CLASS"})
+ * @Attributes(
+ *     @Attribute("accessControl", type="string"),
+ *     @Attribute("accessControlMessage", type="string"),
+ *     @Attribute("attributes", type="array"),
+ *     @Attribute("collectionOperations", type="array"),
+ *     @Attribute("denormalizationContext", type="array"),
+ *     @Attribute("description", type="string"),
+ *     @Attribute("fetchPartial", type="bool"),
+ *     @Attribute("forceEager", type="bool"),
+ *     @Attribute("filters", type="string[]"),
+ *     @Attribute("graphql", type="array"),
+ *     @Attribute("iri", type="string"),
+ *     @Attribute("itemOperations", type="array"),
+ *     @Attribute("maximumItemsPerPage", type="int"),
+ *     @Attribute("normalizationContext", type="array"),
+ *     @Attribute("order", type="array"),
+ *     @Attribute("paginationClientEnabled", type="bool"),
+ *     @Attribute("paginationClientItemsPerPage", type="bool"),
+ *     @Attribute("paginationClientPartial", type="bool"),
+ *     @Attribute("paginationEnabled", type="bool"),
+ *     @Attribute("paginationFetchJoinCollection", type="bool"),
+ *     @Attribute("paginationItemsPerPage", type="int"),
+ *     @Attribute("paginationPartial", type="bool"),
+ *     @Attribute("routePrefix", type="string"),
+ *     @Attribute("shortName", type="string"),
+ *     @Attribute("subresourceOperations", type="array"),
+ *     @Attribute("validationGroups", type="mixed")
+ * )
```

Comment:
FTR I'm using vim and all I see is noise :smiling_imp:

Reply:
Dear @soyuka,

I encourage you to download a real IDE. Here is the link to PHPStorm: https://www.jetbrains.com/phpstorm/download/.

I hope this could help you to find the truth!

Yours sincerely.

Reply:
More seriously, `@Attribute` annotations are needed when using a constructor in your annotation class. The noise here is the private properties.